test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1

```diff
@@ -117,16 +166,153 @@ Describe "Send-MailMessage" -Tags CI, RequireSudoOnUnix {
+        $mail.Headers["To"].Split(", ") | Should -BeExactly $InputObject.To
+        If($mail.Headers["Cc"])
+        {
+            $mail.Headers["Cc"].Split(", ") | Should -BeExactly $InputObject.Cc
+        }
+        If($mail.Headers["Bcc"])
+        {
+            $mail.Headers["Bcc"].Split(", ") | Should -BeExactly $InputObject.Bcc
         }
+        If($mail.Headers["Reply-To"])
+        {
+            $mail.Headers["Reply-To"] | Should -BeExactly $InputObject.ReplyTo
+        }
+        $mail.Headers["Subject"] | Should -BeExactly $InputObject.Subject
 
         $mail.MessageParts.Count | Should -BeExactly 1
         $mail.MessageParts[0].BodyData | Should -BeExactly $InputObject.Body
     }
 }
+
+Describe "Send-MailMessage Feature Tests" -Tags Feature, RequireSudoOnUnix {
+    BeforeEach {
+        $server = [netDumbster.smtp.SimpleSmtpServer]::Start(25)
+
+        function Read-Mail
+        {
+            param()
+
+            if($server)
+            {
+                return $server.ReceivedEmail[0]
+            }
+            return $null
+        }
+    }
+
+    AfterEach {
+        if($server)
+        {
+            $server.Stop()
+        }
+    }
+
+    $InputObject = @{
+        From = "user01@example.com"
+        To = "user02@example.com"
+        Subject = "Subject $(Get-Date)"
+        Body = "Body $(Get-Date)"
+        SmtpServer = "127.0.0.1"
+    }
+
+    It "Can send mail message using custom port 2525" {
+        $server.Stop()
+        $customPortServer = [netDumbster.smtp.SimpleSmtpServer]::Start(2525)
+
+        $customPortServer | Should -Not -Be $null
+        $customPortServer.ReceivedEmailCount | Should -BeExactly 0
+
+        Send-MailMessage @InputObject -Port 2525 -ErrorAction SilentlyContinue
+
+        $customPortServer.ReceivedEmailCount | Should -BeExactly 1
+        $customPortServer.Stop()
+    }
+
+    It "Can throw on wrong mail addresses" {
+        $server | Should -Not -Be $null
+
+        $obj = $InputObject.Clone()
+        $obj.To = "not_a_valid_mail.address"
+
+        { Send-MailMessage @obj -ErrorAction Stop } | Should -Throw -ErrorId "FormatException,Microsoft.PowerShell.Commands.SendMailMessage"
+    }
+
+    It "Can send mail with free-form email address" {
+        $server | Should -Not -Be $null
+
+        $obj = $InputObject.Clone()
+        $obj.From = "User01 <user01@example.com>"
+        $obj.To = "User02 <user02@example.com>"
+
+        Send-MailMessage @obj -ErrorAction SilentlyContinue
+
+        $mail = Read-Mail
+
+        $mail.FromAddress | Should -BeExactly "user01@example.com"
+        $mail.ToAddresses | Should -BeExactly "user02@example.com"
+    }
+
+    It "Can send mail with high priority" {
+        $server | Should -Not -Be $null
+
+        Send-MailMessage @InputObject -Priority High -ErrorAction SilentlyContinue
+
+        $mail = Read-Mail
+        $mail.Priority | Should -BeExactly "urgent"
+    }
+
+    It "Can send mail with body as HTML" {
```

Comment:
```suggestion
    It "Can send mail with body containing HTML" {
```

Reply:
Thank you @RDIL! I've renamed it to "Can send mail with HTML content as body" and also added a `Content-Type` check.